gunrock/app/sm/sm_test.cuh

```diff
@@ -87,6 +95,77 @@ void DisplaySolution(T *array, SizeT length) {
 template <typename GraphT, typename VertexT = typename GraphT::VertexT>
 double CPU_Reference(util::Parameters &parameters, GraphT &data_graph,
                      GraphT &query_graph, VertexT *subgraphs) {
+  typedef typename GraphT::SizeT SizeT;
+
+#ifdef BOOST_FOUND
+  using namespace boost;
+  // User defined counting callback
+  template <typename Graph1,
+            typename Graph2>
+  struct vf2_count_callback {
+  vf2_count_callback(const Graph1& graph1, const Graph2& graph2)
+      : graph1_(graph1), graph2_(graph2), count_(0) {
+          count_ptr = &count_;
+      }
+  
+      template<typename CorrespondenceMap1To2,
+               typename CorrespondenceMap2To1>
+      bool operator()(CorrespondenceMap1To2 f, CorrespondenceMap2To1) {
+          // Count number of matches from isomorphism map
+          (*count_ptr)++;
+          return true;
+      }
+  private:
+      const Graph1& graph1_;
+      const Graph2& graph2_;
+      size_t *count_ptr{nullptr};
```

Comment:
I'm not sure I understand why the `count_ptr` variable exists? Why not increment `count_` directly?

Reply:
'Cause boost copy the callback structure multiple times, it's impossible to accumulate count. I tried to set count_ as a static member, but boost doesn't allow me to do that either.

Reply:
I have a script for you if you wanna try it out: https://gist.github.com/Laurawly/7da2bfc2e0610705c7d3db3c57229c11

Reply:
Ah, I see. You should be able to get rid of the `count_ptr` stuff if you use `std::ref` like so `  vf2_subgraph_iso(graph1, graph2, std::ref(callback));`

Reply:
yeah I can do ref as well.